lib/cn/route.h

```diff
@@ -124,4 +126,17 @@ ekgen_destroy(struct ekey_generator *egen);
 size_t
 ekgen_generate(struct ekey_generator *egen, void *ekbuf, size_t ekbufsz, uint32_t nodeoff);
 
+merr_t
+route_node_key_modify(
+    struct route_map  *map,
+    struct route_node *node,
+    const void        *edge_key,
+    uint               edge_klen);
+
+static HSE_ALWAYS_INLINE int
+route_node_keycmp(const struct route_node *node, const void *key, uint klen)
```

Comment:
I don't see any value in adding this function

Reply:
This is just a helper that works on a route_node.

Reply:
Right, but the helper just grabs the key from the route node. What is the value over just doing what this function does at the call site?

Reply:
Ideally struct route_node should have been private to route.c file and we should do that at some point.. The caller needs to be aware of the key buffer management within route map to grab the key correctly.

Reply:
Ok. Can you create a ticket for hiding the implementation and apply it to the right sprint?

Reply:
If you do, measure before and after hiding to see what the impact of hiding is, preferably a test where all the keys and values fit into RAM so that page faults don't hide the difference.  Also, make keys and values small for the same reason...